### PR TITLE
Swagger-diff Output changes

### PR DESCRIPTION
**Pull request by Gurleen-Singh**
https://github.com/zallek/swagger-diff/issues/10

### CODE REVIEW

**Comment by Gurleen-Singh** [comment]
We have made changes in src/rules and workflow/applyrules.
Added all key values to output and previous variables are returned as it is,so that other users can't
be affected,and those who want to use key variables can use them.
Please review this as early as possible.
Thanks.

**Comment by Gurleen-Singh** [comment]
Please ignore all other commits. Our changes are in  "changed rule output"

**Comment by Gurleen-Singh** [comment]
We have made changes in src/rules and workflow/applyrules.
Added all key values to output and previous variables are returned as it is,so that other users can't
be affected,and those who want to use key variables can use them.
Please ignore all other commits. Our changes are in "changed rule output"
Please review this as early as possible.
Thanks.

**Comment by zallek** [comment]
Could you rebase on master pls to see if tests still pass.
I fixed the build on travis (deps conflict) https://github.com/zallek/swagger-diff/pull/12

File: src/rules/break/edit-base-path.js [hunk 1]
```diff
@@ -4,7 +4,11 @@ export default function editBasePath({ kind, path, lhs, rhs }) {
               && path.length === 1
               && path[0] === 'basePath';
   if (match) {
-    return `Base path turned from ${lhs} to ${rhs}`;
+    return {
+      message: `Base path turned from ${lhs} to ${rhs}`,
+      previousBasePath: lhs,
+      CurrentBasePath: rhs
```
Comment (zallek) [inline src/rules/break/edit-base-path.js]: camelCase

File: src/rules/break/edit-host.js [hunk 2]
```diff
@@ -4,7 +4,11 @@ export default function editHost({ kind, path, lhs, rhs }) {
               && path.length === 1
               && path[0] === 'host';
   if (match) {
-    return `Host turned from ${lhs} to ${rhs}`;
+    return {
+      message: `Host turned from ${lhs} to ${rhs}`,
+      previousHost: lhs,
+      CurrentHost: rhs
```
Comment (zallek) [inline src/rules/break/edit-host.js]: camelCase

File: src/rules/break/edit-object-property-type.js [hunk 3]
```diff
@@ -5,9 +5,15 @@ export default function editObjectPropertyType({ kind, path, lhs, rhs }) {
...
-    const objectPath = path.slice(0, -3).join(' ');
+    const objectPath = path.slice(0, -3).join('/');
     const propertyName = path[path.length - 2];
-    return `${objectPath} - Property ${propertyName} type turn from ${lhs} to ${rhs}`;
+    return {
+      message: `${objectPath} - Property ${propertyName} type turn from ${lhs} to ${rhs}`,
+      path: objectPath,
+      Property: propertyName,
+      previousPropertyType: lhs,
+      CurrentPropertyType: rhs
```
Comment (zallek) [inline src/rules/break/edit-object-property-type.js]: camelCase

File: src/workflow/applyRules.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import forEach from 'lodash.foreach';
 
+var extend = require('extend');
```
Comment (zallek) [inline src/workflow/applyRules.js]: missing in `package.json`

**Comment by zallek** [comment]
Thx
Sound great, json output will definitely be more useful this way :)
So
- rebase on master
- update rules tests output payloads
- update your deps with new eslint version (you may see some new lint errors appear)

And we will be good to go.

**Comment by Gurleen-Singh** [comment]
All comments are addressed .
Please have a look.
Thanks.

**Comment by zallek** [comment]
Installation is still failling (might be due to the modif you made in package.json script build:lib)

**Comment by Gurleen-Singh** [comment]
@zallek: It seems the build has passed now. Please rebase the same with master as soon as possible. Let me know if anything else is needed from my side.
Thanks.

File: src/rules/index.js [hunk 5]
```diff
@@ -5,6 +5,7 @@
 
 export default {
   break: {
+    '.DS_St': require('./break/.DS_St.js'),
```
Comment (zallek) [inline src/rules/index.js]: what's that ?

File: src/rules/break/edit-param-type.js [hunk 6]
```diff
@@ -9,7 +9,14 @@ export default function editParamType({ kind, path, lhs, rhs }) {
     const pathId = path[1];
     const method = path[2];
     const paramName = path[4];
-    return `${pathId} (${method}) - Param ${paramName} type turn from from ${lhs} to ${rhs}`;
+    return {
+      message: `${pathId} (${method}) - Param ${paramName} type turn from from ${lhs} to ${rhs}`,
+      path: pathId,
+      method,
+      responseId: paramName,
+      previousparamName: lhs,
```
Comment (zallek) [inline src/rules/break/edit-param-type.js]: camelCase

File: src/rules/break/edit-param-type.js [hunk 7]
```diff
@@ -9,7 +9,14 @@ export default function editParamType({ kind, path, lhs, rhs }) {
...
+      currentparamName: rhs,
```
Comment (zallek) [inline src/rules/break/edit-param-type.js]: camelCase

File: src/rules/break/edit-response.js [hunk 8]
```diff
@@ -10,7 +10,14 @@ export default function editResponse({ kind, path, lhs, rhs }) {
     const pathId = path[1];
     const method = path[2];
     const responseId = path[4];
-    return `${pathId} (${method}) - Response ${responseId} turned from ${lhs} to ${rhs}`;
+    return {
+      message: `${pathId} (${method}) - Response ${responseId} turned from ${lhs} to ${rhs}`,
+      path: pathId,
+      method,
+      responseId,
+      previousresponseId: lhs,
```
Comment (zallek) [inline src/rules/break/edit-response.js]: camelCase

File: src/rules/break/edit-response.js [hunk 9]
```diff
@@ -10,7 +10,14 @@ export default function editResponse({ kind, path, lhs, rhs }) {
...
+      currentresponseId: rhs,
```
Comment (zallek) [inline src/rules/break/edit-response.js]: camelCase

File: package.json [hunk 10]
```diff
@@ -90,4 +92,4 @@
   "bin": {
     "swagger-diff": "bin/swagger-diff.js"
   }
-}
+}
```
Comment (zallek) [inline package.json]: missing new line

File: src/rules/break/add-required-param.js [hunk 11]
```diff
@@ -9,7 +9,12 @@ export default function addRequiredParam({ kind, path, lhs, rhs }) {
     const pathId = path[1];
     const method = path[2];
     const paramName = path[4];
-    return `${pathId} (${method}) - Required param ${paramName} added`;
+    return {
+      message: `${pathId} (${method}) - Required param ${paramName} added`,
+      path: pathId,
+      method,
+      paramAdded: paramName,
```
Comment (zallek) [inline src/rules/break/add-required-param.js]: param

File: src/rules/break/delete-produce.js [hunk 12]
```diff
@@ -8,7 +8,12 @@ export default function deleteProduce({ kind, path, index, item }) {
   if (match) {
     const pathId = path[1];
     const method = path[2];
-    return `${pathId} (${method}) - Produces ${item.lhs} deleted`;
+    return {
+      message: `${pathId} (${method}) - Produces ${item.lhs} deleted`,
+      path: pathId,
+      method,
+      itemDeleted: item.lhs,
```
Comment (zallek) [inline src/rules/break/delete-produce.js]: produce

File: src/rules/break/edit-object-property-type.js [hunk 13]
```diff
@@ -5,9 +5,15 @@ export default function editObjectPropertyType({ kind, path, lhs, rhs }) {
...
   if (match) {
-    const objectPath = path.slice(0, -3).join(' ');
+    const objectPath = path.slice(0, -3).join('/');
     const propertyName = path[path.length - 2];
-    return `${objectPath} - Property ${propertyName} type turn from ${lhs} to ${rhs}`;
+    return {
+      message: `${objectPath} - Property ${propertyName} type turn from ${lhs} to ${rhs}`,
+      path: objectPath,
+      Property: propertyName,
+      previousPropertyType: lhs,
```
Comment (zallek) [inline src/rules/break/edit-object-property-type.js]: previousType

File: src/rules/break/edit-object-property-type.js [hunk 14]
```diff
@@ -5,9 +5,15 @@ export default function editObjectPropertyType({ kind, path, lhs, rhs }) {
...
+      currentPropertyType: rhs,
```
Comment (zallek) [inline src/rules/break/edit-object-property-type.js]: currentType

File: src/rules/break/edit-param-required.js [hunk 15]
```diff
@@ -10,7 +10,12 @@ export default function editParamRequired({ kind, path, lhs, rhs }) {
     const pathId = path[1];
     const method = path[2];
     const paramName = path[4];
-    return `${pathId} (${method}) - Param ${paramName} became required`;
+    return {
+      message: `${pathId} (${method}) - Param ${paramName} became required`,
+      path: pathId,
+      method,
+      paramName,
```
Comment (zallek) [inline src/rules/break/edit-param-required.js]: param

File: src/rules/break/edit-param-type.js [hunk 16]
```diff
@@ -9,7 +9,14 @@ export default function editParamType({ kind, path, lhs, rhs }) {
     const pathId = path[1];
     const method = path[2];
     const paramName = path[4];
-    return `${pathId} (${method}) - Param ${paramName} type turn from from ${lhs} to ${rhs}`;
+    return {
+      message: `${pathId} (${method}) - Param ${paramName} type turn from from ${lhs} to ${rhs}`,
+      path: pathId,
+      method,
+      responseId: paramName,
```
Comment (zallek) [inline src/rules/break/edit-param-type.js]: param

File: src/rules/index.js [hunk 17]
```diff
@@ -25,6 +26,7 @@ export default {
     'edit-response': require('./break/edit-response.js'),
   },
   smooth: {
+    '.DS_St': require('./smooth/.DS_St.js'),
```
Comment (zallek) [inline src/rules/index.js]: what's that ?

File: src/rules/smooth/add-description.js [hunk 18]
```diff
@@ -6,7 +6,12 @@ export default function editDescription({ kind, path, lhs, rhs }) {
   if (match) {
     const pathId = path[1];
     const method = path[2];
-    return `${pathId} (${method}) - Description added: ${rhs}`;
+    return {
+      message: `${pathId} (${method}) - Description added: ${rhs}`,
+      path: pathId,
+      method,
+      descriptionAdded: rhs,
```
Comment (zallek) [inline src/rules/smooth/add-description.js]: description

File: src/rules/smooth/add-optional-object-property.js [hunk 19]
```diff
@@ -5,9 +5,13 @@ export default function addOptionalObjectProperty({ kind, path, lhs, rhs }) {
...
               && !rhs.required;
   if (match) {
-    const objectPath = path.slice(0, -2).join(' ');
+    const objectPath = path.slice(0, -2).join('/');
     const propertyName = path[path.length - 1];
-    return `${objectPath} - Optional property ${propertyName} added`;
+    return {
+      message: `${objectPath} - Optional property ${propertyName} added`,
+      path: objectPath,
+      propertyName,
```
Comment (zallek) [inline src/rules/smooth/add-optional-object-property.js]: property

File: src/rules/smooth/add-optional-param.js [hunk 20]
```diff
@@ -9,7 +9,12 @@ export default function addOptionalParam({ kind, path, lhs, rhs }) {
     const pathId = path[1];
     const method = path[2];
     const paramName = path[4];
-    return `${pathId} (${method}) - Optional param ${paramName} added`;
+    return {
+      message: `${pathId} (${method}) - Optional param ${paramName} added`,
+      path: pathId,
+      method,
+      paramName,
```
Comment (zallek) [inline src/rules/smooth/add-optional-param.js]: param

File: src/rules/smooth/delete-object-property.js [hunk 21]
```diff
@@ -4,9 +4,13 @@ export default function deleteObjectProperty({ kind, path, lhs, rhs }) {
...
               && path[path.length - 2] === 'properties';
   if (match) {
-    const objectPath = path.slice(0, -2).join(' ');
+    const objectPath = path.slice(0, -2).join('/');
     const propertyName = path[path.length - 1];
-    return `${objectPath} - Property ${propertyName} deleted`;
+    return {
+      message: `${objectPath} - Property ${propertyName} deleted`,
+      path: objectPath,
+      propertyName,
```
Comment (zallek) [inline src/rules/smooth/delete-object-property.js]: property

File: src/rules/smooth/delete-param.js [hunk 22]
```diff
@@ -8,7 +8,12 @@ export default function deleteParam({ kind, path, lhs, rhs }) {
     const pathId = path[1];
     const method = path[2];
     const paramName = path[4];
-    return `${pathId} (${method}) - Param ${paramName} deleted`;
+    return {
+      message: `${pathId} (${method}) - Param ${paramName} deleted`,
+      path: pathId,
+      method,
+      parameterName: paramName,
```
Comment (zallek) [inline src/rules/smooth/delete-param.js]: param

File: src/rules/smooth/edit-description.js [hunk 23]
```diff
@@ -6,7 +6,13 @@ export default function editDescription({ kind, path, lhs, rhs }) {
   if (match) {
     const pathId = path[1];
     const method = path[2];
-    return `${pathId} (${method}) - Description turned from ${lhs} to ${rhs}`;
+    return {
+      message: `${pathId} (${method}) - Description turned from ${lhs} to ${rhs}`,
+      path: pathId,
+      method,
+      previousName: lhs,
```
Comment (zallek) [inline src/rules/smooth/edit-description.js]: previousDescription

File: src/rules/smooth/edit-description.js [hunk 24]
```diff
@@ -6,7 +6,13 @@ export default function editDescription({ kind, path, lhs, rhs }) {
...
+      currentName: rhs,
```
Comment (zallek) [inline src/rules/smooth/edit-description.js]: currentDescription

**Comment by zallek** [comment]
@Gurleen-Singh 
I've made a bunch of comments on rules payload interface. Because it has to be done right since the beginning to avoid future breaking changes :)
So I tried to standardize the names and fix some of them

**Comment by Gurleen-Singh** [comment]
Okay.
Let me know if I can help or once it's done.

File: src/workflow/applyRules.js [hunk 25]
```diff
@@ -1,5 +1,6 @@
 import forEach from 'lodash.foreach';
 
+const extend = require('extend');
```
Comment (zallek) [inline src/workflow/applyRules.js]: import

**Comment by zallek** [comment]
moved to https://github.com/zallek/swagger-diff/pull/14

**Pull request closed without merging**